File: AVc/app/src/main/java/com/hfad/avc/HelperModule.java
package com.hfad.avc;

import androidx.room.Room;

import com.hfad.avc.ui.database.AppDatabase;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class HelperModule {
    AppDatabase database;

    @Provides
    @Singleton
    AppDatabase getDatabase() {
        //Избавиться от контакста в виде INSTANCE
        database = Room.databaseBuilder(Applications.INSTANCE, AppDatabase.class, "database")
                .allowMainThreadQueries()
                .build();
        return database;
    }

}

File: app/src/main/java/com/hfad/avc/dagger/module/HelperModule.kt
package com.hfad.avc.dagger.module


import android.content.Context
import androidx.room.Room
import com.hfad.avc.data.database.AppDatabase
import com.hfad.avc.managers.DBManager
import dagger.Module
import dagger.Provides
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton


@Module
class HelperModule constructor(
   var database: AppDatabase? = null
) {

   @JvmName("getDatabase1")
   @Provides
   @Singleton
   fun getDatabase(context: Context): AppDatabase {
      if (database == null) {
         //Избавиться от контакста в виде INSTANCE
         database =
            Room.databaseBuilder(context, AppDatabase::class.java, "database_db1")
               .allowMainThreadQueries()
               .fallbackToDestructiveMigration()
               .build()
      }
      return database as AppDatabase
   }

   @Provides
   @Singleton
   fun getDBManager(): DBManager {
      return DBManager()
   }


}